internal/controlplane/handlers_authz.go

```diff
@@ -265,12 +265,121 @@ func (*Server) ListRoleAssignments(
 	return nil, nil
 }
 
-// AssignRole assigns a role to a user on a project
-func (*Server) AssignRole(context.Context, *minder.AssignRoleRequest) (*minder.AssignRoleResponse, error) {
-	return nil, nil
+// AssignRole assigns a role to a user on a project.
+// Note that this assumes that the request has already been authorized.
+func (s *Server) AssignRole(ctx context.Context, req *minder.AssignRoleRequest) (*minder.AssignRoleResponse, error) {
+	// Request Validation
+	role := req.GetRoleAssignment().GetRole()
+	sub := req.GetRoleAssignment().GetSubject()
+
+	if role == "" || sub == "" {
+		return nil, util.UserVisibleError(codes.InvalidArgument, "role and subject must be specified")
+	}
+
+	authzrole, err := authz.ParseRole(role)
+	if err != nil {
+		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
+	}
+
+	// Determine target project.
+	entityCtx := engine.EntityFromContext(ctx)
+	currentProjectID := entityCtx.Project.ID
+	targetProjectID := currentProjectID
```

Comment:
I'm wondering if there's any benefit in differentiating the current project and the target project. 
If we have a command like: `minder role assign --subject user1 --project p1`, then the project in the context (i.e current project) would also be `p1`. At least that's how we do it for other commands.

Reply:
I think you're right. Let me remove this logic from here.

Reply:
done